backend/src/main/java/com/google/models/Anomaly.java

```diff
@@ -0,0 +1,150 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.models;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.models.MetricValue;
+import com.google.models.Timestamp;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import java.util.Map;
+import java.util.HashMap;
+
+/** Store anomaly-related data. */
+public final class Anomaly {
+  public static final String ANOMALY_ENTITY_KIND = "anomaly";
+  public static final String METRIC_NAME_PROPERTY = "metricName";
+  public static final String DIMENSION_NAME_PROPERTY = "dimensionName";
+  public static final String DATA_POINTS_PROPERTY = "dataPoints";
+
+  private static final String DUMMY_METRIC_NAME = "Sample metric name";
+  private static final String DUMMY_DIMENSION_NAME = "Sample dimension name";
+  private static final Map<Timestamp, MetricValue> DUMMY_DATA_POINTS = ImmutableMap.of( 
+      Timestamp.getDummyTimestamp(1), new MetricValue(1), 
+      Timestamp.getDummyTimestamp(2), new MetricValue(2), 
+      Timestamp.getDummyTimestamp(3), new MetricValue(3));
+  
+  private final Timestamp timestampDate;
+  private final String metricName;
+  private final String dimensionName;
+  private final Map<Timestamp, MetricValue> dataPoints;
+  
+  public Anomaly(Timestamp timestampDate, String metricName, String dimensionName, 
+      Map<Timestamp, MetricValue> dataPoints) {
+    this.timestampDate = timestampDate;
+    this.metricName = metricName;
+    this.dimensionName = dimensionName;
+    this.dataPoints = dataPoints;
+  }
+
+  public Timestamp getTimestamp() {
+    return timestampDate;
+  }
+
+  public String getMetricName() {
+    return metricName;
+  }
+
+  public String getDimensionName() {
+    return dimensionName;
+  }
+
+  public Map<Timestamp, MetricValue> getDataPoints() {
+    return dataPoints;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
also have a check for if (o == null) return false;
also I think you can inline all of these short if statements if you want to make it shorter

Reply:
this is a good point, i added a check for it in the unit test for `equals()` for all our models :) thanks!

but i don't think it is necessary to check `if (o == null)` because i think `if (!(o instanceof Anomaly))` should cover for that...